### PR TITLE
Feat/routes list fe to be

### PR DESCRIPTION
**Pull request by oyakovytskyi**
## Describe your changes

- Created and refactored several components and logic to improve the route page.
- Added functionality to call API endpoint for fetch routes and manage the data using Redux.
- Introduced date range selection from the calendar and implemented driver name search functionality.
- Refined UI components, including a dropdown with checkboxes for route selection.
- Consolidated all page logic into a custom hook for better code maintainability and structure.

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://codecraftersintership.atlassian.net/browse/SCRUM-54)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

https://www.loom.com/share/8b893ed721da426d8a3358c8d9b21bc5?sid=fbbafbc6-3f33-4447-878c-0f1f01f38efa

### CODE REVIEW

File: src/pages/Routes/components/NoRoutesMessage/index.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { Typography } from '@mui/material';
+
+const NoRoutesMessage: React.FC = () => (
+  <Typography variant="body1" align="center" sx={{ padding: 4 }}>
+    No routes have been sent to drivers yet
```
Comment (asdelay) [inline src/pages/Routes/components/NoRoutesMessage/index.tsx]: i18n

**Review by asdelay — COMMENTED** [review]
fix i18 and i approve

**Review by asdelay — APPROVED** [review]

**Review by asdelay — APPROVED** [review]

File: src/components/CalendarRange/index.tsx [hunk 2]
```diff
@@ -5,11 +5,23 @@ import { DateRangePicker } from '@mui/x-date-pickers-pro/DateRangePicker';
...
+import { Dayjs } from 'dayjs';
 
-const CalendarRange: React.FC = () => {
+const CalendarRange: React.FC<{
+  onDateChange: (start: string, end: string) => void;
+}> = ({ onDateChange }) => {
+  const handleDateChange = (range: [Dayjs | null, Dayjs | null]) => {
+    if (range[0] && range[1]) {
+      onDateChange(
+        range[0].format('YYYY-MM-DD'),
```
Comment (lyudad) [inline src/components/CalendarRange/index.tsx]: 1.1.13

File: src/constants/days.ts
Comment (lyudad) [inline src/constants/days.ts:1]: use one constants files to everything connected with dates

File: src/constants/numbers.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export const NOT_ASSIGNED_ORDERS_BLOCK = -1;
```
Comment (lyudad) [inline src/constants/numbers.ts]: why this in general constans, not in the local

File: src/pages/Routes/utils/index.ts
Comment (lyudad) [inline src/pages/Routes/utils/index.ts:1]: it should be general utils file, not under Routes, as date formating could be used across the project

File: src/components/CalendarRange/index.tsx (hunk 2 above)
Comment (oyakovytskyi) [reply src/components/CalendarRange/index.tsx]: done

File: src/pages/Routes/utils/index.ts
Comment (oyakovytskyi) [reply src/pages/Routes/utils/index.ts:1]: make changes

File: src/constants/numbers.ts (hunk 3 above)
Comment (oyakovytskyi) [reply src/constants/numbers.ts]: done

**Review by asdelay — APPROVED** [review]

**Review by lyudad — APPROVED** [review]

**Pull request merged**